engine/pkg/orm/client.go

```diff
@@ -119,15 +120,18 @@ type ResourceClient interface {
 	CreateResource(ctx context.Context, resource *resourcemeta.ResourceMeta) error
 	UpsertResource(ctx context.Context, resource *resourcemeta.ResourceMeta) error
 	UpdateResource(ctx context.Context, resource *resourcemeta.ResourceMeta) error
-	DeleteResource(ctx context.Context, resourceID string) (Result, error)
-	GetResourceByID(ctx context.Context, resourceID string) (*resourcemeta.ResourceMeta, error)
+	GetResourceByID(ctx context.Context, jobID engineModel.JobID, resourceID string) (*resourcemeta.ResourceMeta, error)
```

Comment:
Is it better to use ` GetResourceByKey(ctx context.Context, key resourcemeta.ResourceKey)`

Reply:
Sure. I will unify the usage of `ResourceKey` .